common

```diff
@@ -24,9 +24,107 @@ cli_log() {
     exit 130
 }
 
+
+# Status Types
+TYPE_SUCCESS='SUCCESS'
+TYPE_INFO='INFO'
+TYPE_ERROR='ERROR'
+# Status Type Colors
+success_color="\033[1;32m"  # Green color
+info_color="\033[1;33m"  # Blue color
+error_color="\033[1;31m"  # Red color
+reset_color="\033[0m"    # Reset color
+
+# calling syntax : status_log <type> <message>
+# supported types : "info", "success", "error"
+status_log() {
+
+  # first select color
+  log_color=""
+  if [ $1 == $TYPE_INFO ]; then log_color=$info_color
+  elif [ $1 == $TYPE_SUCCESS ]; then log_color=$success_color
+  elif [ $1 == $TYPE_ERROR ]; then log_color=$error_color
+  fi
+
+  gum log "$(printf "${log_color}${2}${reset_color}")"
+
+}
+
+# check and install all pre-requsite packages
+install_prerequisite(){
+
+
+  # if OS is gnu, we want to check and install snap
+  # this package will be used to install other packages in future
+  if [ $OS == "gnu" ] && ! snap version >/dev/null 2>&1; then
+    sudo apt update
+    sudo apt install snapd
+  fi
+
+
+  # installing helm
+  if ! helm version >/dev/null 2>&1; then
+    echo "helm is a prerequisite for this script to run."
+    echo "installing helm..."
+    if [ $OS == "mac" ]; then 
+      brew install helm
+    elif [ $OS == "gnu" ]; then
+      sudo snap install helm --classic
+    else
+      errMsg="Failed to install a prerequisite helm, OS ${OS} not supported at the moment"
+      echo $errMsg
+      status_log $TYPE_ERROR $errMsg
+      exit 1
+    fi
+  fi
+
+  if ! kubectl version >/dev/null 2>&1; then
+    echo "kubectl is a prerequisite for this script to run."
+    echo "installing kubectl..."
+    if [ $OS == "mac" ]; then 
+      brew install kubectl
+    elif [ $OS == "gnu" ]; then
+      sudo snap install kubectl --classic
+    else
+      errMsg="Failed to install a prerequisite kubectl, OS ${OS} not supported at the moment"
+      echo $errMsg
+      status_log $TYPE_ERROR $errMsg
+      exit 1
+    fi
+  fi
+
+  # Ensure gum is installed: https://github.com/charmbracelet/gum
+  if ! gum --version >/dev/null 2>&1; then
+    echo "gum is a prerequisite for this script to run."
+    echo "installing gum..."
+    if [ "${OS}" == "mac" ]; then
+      brew install gum
+    elif [ "${OS}" == "gnu" ]; then 
+      sudo mkdir -p /etc/apt/keyrings
+      curl -fsSL https://repo.charm.sh/apt/gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/charm.gpg
+      echo "deb [signed-by=/etc/apt/keyrings/charm.gpg] https://repo.charm.sh/apt/ * *" | sudo tee /etc/apt/sources.list.d/charm.list
+      sudo apt update && sudo apt install gum
+      sudo ln -s /usr/bin/gum /snap/bin/gum
+    else
+      errMsg="Failed to install a prerequisite gum, OS ${OS} not supported at the moment"
+      echo $errMsg
+      status_log $TYPE_ERROR $errMsg
+      exit 1
+    fi
+  fi
+}
+
+# need to create log folder as its a pre-requsite
+create_logs_dir(){
+  if ! ls $GRPL_WORKDIR | grep 'logs' >/dev/null 2>&1; then
+    mkdir logs
```

Comment:
@zahid-iqbal-alpha 
I think this command is not good enough like this.
it will create the logs folder in the current folder.
when running it as a brew package, this may mess up things as it creates logs directories, wherever you currently are located.
can you please fix that?
thanks

Reply:
Hi @grapple-solutions 
I added this logs folder creation code after viewing this code sinipit where we are trying to push the logs to 
$GRPL_WORKDIR/logs/ 
`    "$GRPL_WORKDIR/commands/install" $(echo $@ | sed 's,install,,g' | sed 's,i ,,') | tee -ia "$GRPL_WORKDIR/logs/install.log"  | grep $uuid`

So if not here, then where do you suggest we store the logs?